### PR TITLE
EZP-31594: Refactored --siteaccess command option to not rely on custom Application class

### PR DESCRIPTION
**Pull request by webhdx**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | EZP-31594
| **Type**                                   | improvement
| **Target eZ Platform version** | `v3.3`
| **BC breaks**                          | no
| **Doc needed**                       | no

This removes custom `Application` class completely as well as `CONSOLE_INIT` event but I assume it can be safely removed because it has been marked `@internal`. It injects siteaccess option via `ConsoleCommandPass`. Maybe it could be improved and only add the option to commands marked with particular tag but for now I reflected the original behavior.

There is complementary PR in meta: https://github.com/ezsystems/ezplatform/pull/627

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [x] Provided automated test coverage.
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [x] Asked for a review (ping `@ezsystems/php-dev-team`).

### CODE REVIEW

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/ConsoleCommandPass.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class ConsoleCommandPass implements CompilerPassInterface
```
Comment (adamwojs) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/ConsoleCommandPass.php:13]:
```suggestion
final class ConsoleCommandPass implements CompilerPassInterface
```

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/ConsoleCommandPass.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
```
Comment (adamwojs) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/ConsoleCommandPass.php:6]:
```suggestion
/**
 * @copyright Copyright (C) eZ Systems AS. All rights reserved.
 * @license For full copyright and license information view LICENSE file distributed with this source code.
 */
declare(strict_types=1);
```

File: eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Compiler/ConsoleCommandPassTest.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Tests\DependencyInjection\Compiler;
+
+use eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler\ConsoleCommandPass;
+use Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractCompilerPassTestCase;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+class ConsoleCommandPassTest extends AbstractCompilerPassTestCase
```
Comment (adamwojs) [inline eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Compiler/ConsoleCommandPassTest.php:15]:
```suggestion
final class ConsoleCommandPassTest extends AbstractCompilerPassTestCase
```

**Comment by webhdx** [comment]
Thanks @adamwojs. Code fixed according to your review. Tests are now passing, I had to modify test suite to use my fork of meta repository.

**Review by ViniTou — APPROVED** [review]

**Review by adamwojs — APPROVED** [review]

**Pull request merged**